pkg/storage/internal/actions/internalize.go

```diff
@@ -62,6 +59,17 @@ func (in *internalize) Internalize(ctx context.Context, userID int, args *wdk.In
 		return nil, fmt.Errorf("failed to generate random reference: %w", err)
 	}
 
+	// TODO: Don't upsert ProvenTxReq if the transaction is already known in ProvenTx (not *Req)
+	err = in.provenTxRepo.UpsertProvenTxReq(ctx, &entity.UpsertProvenTxReq{
+		TxID:      txID,
+		RawTx:     tx.Bytes(),
+		InputBeef: args.Tx,
+		Status:    wdk.ProvenTxStatusUnmined,
+	}, entity.InternalizeActionHistoryNote, entity.UserIDHistoryAttr(userID))
```

Comment:
maybe package name `history` would be clearer

Reply:
I know, but it's just about a package name, which change should be simple at this stage (and maybe we just move later this package and that's all).